ext/pgsql/pgsql.c

```diff
@@ -1057,15 +1057,13 @@ PHP_FUNCTION(pg_query)
 
 static void _php_pgsql_free_params(char **params, int num_params)
 {
-	if (num_params > 0) {
-		int i;
-		for (i = 0; i < num_params; i++) {
-			if (params[i]) {
-				efree(params[i]);
-			}
+	int i;
+	for (i = 0; i < num_params; i++) {
+		if (params[i]) {
+			efree(params[i]);
 		}
-		efree(params);
 	}
+	efree(params);
```

Comment:
was it the "point of contention" ? in that case you can just change it place.

Reply:
if num_params == 0 then params array was never freed

Reply:
Yes, e.g. if you exit here in the first iteration when i==0, then `params` is never freed:

https://github.com/php/php-src/blob/28ce1b04ef2b818ccc64df656865ec20af2f35fc/ext/pgsql/pgsql.c#L1130-L1131

Removing the `num_params > 0` check fixes it. It is a useless check anyway: the loop will already check this and we should not do the check for freeing `params`.

Reply:
sure what I meant originally was just putting the ```efree(params);``` out of the if block would have worked as well but that s detail.